### PR TITLE
test: increased unit tests coverage for /handlers/kafka.go

### PR DESCRIPTION
**Pull request by valeriiashapoval**
## Description
Increase unit test code coverage in internal/kafka/internal/handlers/kafka.go - (https://issues.redhat.com/browse/MGDSTRM-8765)

## Verification Steps
make test
make test/html/coverage/report

## Checklist (Definition of Done)
- [X] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [X] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

File: internal/kafka/internal/handlers/kafka_test.go [hunk 1]
```diff
@@ -0,0 +1,591 @@
...
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			h := NewKafkaHandler(tt.fields.service, tt.fields.providerConfig, tt.fields.authService, tt.fields.kafkaConfig)
+			req, rw := GetHandlerParams("GET", "/{id}", nil)
+			req = mux.SetURLVars(req, map[string]string{"id": id})
+			h.Get(rw, req)
+			resp := rw.Result()
+			Expect(resp.StatusCode).To(Equal(tt.wantStatusCode))
+			resp.Body.Close()
```
Comment (ziccardi) [inline internal/kafka/internal/handlers/kafka_test.go:86]: If the `Expect` would fail, the test would exit without closing the body. Move this before the `Expect`

File: internal/kafka/internal/handlers/kafka_test.go [hunk 2]
```diff
@@ -0,0 +1,591 @@
...
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			h := NewKafkaHandler(tt.fields.service, tt.fields.providerConfig, tt.fields.authService, tt.fields.kafkaConfig)
+			req, rw := GetHandlerParams("DELETE", tt.args.url, nil)
+			h.Delete(rw, req)
+			resp := rw.Result()
+			Expect(resp.StatusCode).To(Equal(tt.wantStatusCode))
+			resp.Body.Close()
```
Comment (ziccardi) [inline internal/kafka/internal/handlers/kafka_test.go:154]: See previous comment

File: internal/kafka/internal/handlers/kafka_test.go [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			h := NewKafkaHandler(tt.fields.service, tt.fields.providerConfig, tt.fields.authService, tt.fields.kafkaConfig)
+			req, rw := GetHandlerParams("GET", tt.args.url, nil)
+			h.List(rw, req)
+			resp := rw.Result()
+			Expect(resp.StatusCode).To(Equal(tt.wantStatusCode))
+			resp.Body.Close()
```
Comment (ziccardi) [inline internal/kafka/internal/handlers/kafka_test.go:244]: See previous comment

File: internal/kafka/internal/handlers/kafka_test.go [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			h := NewKafkaHandler(tt.fields.service, tt.fields.providerConfig, tt.fields.authService, tt.fields.kafkaConfig)
+			req, rw := GetHandlerParams("PATCH", tt.args.url, bytes.NewBuffer(tt.args.body))
+			req = req.WithContext(tt.args.ctx)
+			h.Update(rw, req)
+			resp := rw.Result()
+			Expect(resp.StatusCode).To(Equal(tt.wantStatusCode))
+			resp.Body.Close()
```
Comment (ziccardi) [inline internal/kafka/internal/handlers/kafka_test.go:359]: See previous comment

File: internal/kafka/internal/handlers/kafka_test.go [hunk 5]
```diff
@@ -0,0 +1,591 @@
...
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			h := NewKafkaHandler(tt.fields.service, tt.fields.providerConfig, tt.fields.authService, tt.fields.kafkaConfig)
+			req, rw := GetHandlerParams("CREATE", tt.args.url, bytes.NewBuffer(tt.args.body))
+			req = req.WithContext(tt.args.ctx)
+			h.Create(rw, req)
+			resp := rw.Result()
+			Expect(resp.StatusCode).To(Equal(tt.wantStatusCode))
+			resp.Body.Close()
```
Comment (ziccardi) [inline internal/kafka/internal/handlers/kafka_test.go:588]: See previous comment

File: internal/kafka/internal/handlers/kafka_test.go (hunk 1 above)
Comment (valeriiashapoval) [reply internal/kafka/internal/handlers/kafka_test.go:85]: > If the `Expect` would fail, the test would exit without closing the body. Move this before the `Expect`

Thank you @ziccardi ! Just pushed the changes.

File: internal/kafka/internal/handlers/cloud_providers_test.go [hunk 6]
```diff
@@ -15,7 +15,7 @@ import (
 )
 
 var (
-	supportedProviders = config.ProviderConfig{
+	SupportedProviders = config.ProviderConfig{
```
Comment (machi1990) [inline internal/kafka/internal/handlers/cloud_providers_test.go:18]: why is this made public? i.e why the change from small `s` to capital `S`?

File: internal/kafka/internal/handlers/cloud_providers_test.go [hunk 7]
```diff
@@ -33,7 +33,7 @@ var (
 			},
 		},
 	}
-	fullKafkaConfig = config.KafkaConfig{
+	FullKafkaConfig = config.KafkaConfig{
```
Comment (machi1990) [inline internal/kafka/internal/handlers/cloud_providers_test.go:36]: why is this made public?

**Review by machi1990 — CHANGES_REQUESTED** [review]

File: internal/kafka/internal/handlers/cloud_providers_test.go (hunk 6 above)
Comment (valeriiashapoval) [reply internal/kafka/internal/handlers/cloud_providers_test.go]: > why is this made public? i.e why the change from small `s` to capital `S`?

@machi1990 was mainly a mistake, I have reverted those changes, thanks!

**Review by machi1990 — APPROVED** [review]
lgtm
Thanks @valeriiashapoval

**Review by ziccardi — APPROVED** [review]
LGTM
